x-pack/plugins/observability_solution/apm/server/routes/entities/get_entities.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { kqlQuery } from '@kbn/observability-plugin/server';
+import { ENTITY_ENVIRONMENT, FIRST_SEEN, LAST_SEEN } from '../../../common/es_fields/entities';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { EntitiesESClient } from '../../lib/helpers/create_es_client/create_assets_es_client/create_assets_es_clients';
+import { EntitiesRaw, ServiceEntities } from './types';
+
+export function entitiesRangeQuery(start: number, end: number): QueryDslQueryContainer[] {
+  return [
+    {
+      range: {
+        [FIRST_SEEN]: {
+          gte: start,
+        },
+      },
+    },
+    {
+      range: {
+        [LAST_SEEN]: {
+          lte: end,
+        },
+      },
+    },
+  ];
+}
+
+export async function getEntities({
+  entitiesESClient,
+  start,
+  end,
+  environment,
+  kuery,
+  size,
+}: {
+  entitiesESClient: EntitiesESClient;
+  start: number;
+  end: number;
+  environment: string;
+  kuery: string;
+  size: number;
+}) {
+  const entities = (
+    await entitiesESClient.search(`get_entities`, {
+      body: {
+        size,
+        track_total_hits: false,
+        _source: ['agent.name', 'entity', 'data_stream'],
```

Comment:
nit: Looks like we could use the `AGENT_NAME` constant here. Also, dhould we create [constants](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/observability_shared/common/field_names/elasticsearch.ts)  for `entity` and `data_steam.

I also noticed that the only field from `data_stream` we need is the `type`, so would `data_steam.type` make more sense here?